schema/validator.go

```diff
@@ -44,6 +50,13 @@ func (v Validator) Validate(src io.Reader) error {
 		return errors.Wrap(err, "unable to read the document file")
 	}
 
+	if f := mapValidateDescendants[v]; f != nil {
```

Comment:
@wking ,
In the case of setting `nil` to some key, the checking by `ok` will have risk.
`f` is `nil` when `ok` is `true`, the next calling `f` will lead to panic.
So I think it is safe to check `f` is not `nil`.